plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Library.java

```
@@ -102,4 +104,14 @@ public void setLibraryDependencies(List<String> libraryDependencies) {
       this.libraryDependencies = new ArrayList<>(libraryDependencies);
     }
   }
+
+  public void setRecommendation(LibraryRecommendation recommendation) {
+    if (recommendation != null) {
+      this.recommendation = recommendation;
```

This doesn't clear it though. Maybe we should throw the exception?